lib/core/src/sdk.rs

```diff
@@ -3349,20 +3349,34 @@ impl LiquidSdk {
             None => None,
         };
 
-        let lnurl_pay_domain = match prepare_response.data.ln_address {
-            Some(_) => None,
-            None => Some(prepare_response.data.domain),
-        };
         if let (Some(tx_id), Some(destination)) =
             (payment.tx_id.clone(), payment.destination.clone())
         {
+            // Uses existing `ln_address` if it's available, otherwise fallback to `ln_address` from `metadata_str`
+            let ln_address = derive_ln_address(
```

Comment:
What is the use case for searching in the metadata? Isn't data.ln_address enough?

Reply:
I can't reproduce it but there was a case where `data.ln_address` was missing, but the [internet identifier](https://github.com/lnurl/luds/blob/luds/16.md) was available on the `metadata_str` on a payment to WoS ln address.

Either way, this looks like something to be handled on the input parser itself.

Reply:
Removed and reduced the scope of the PR to only extract `description` from `metadata_str`.